src/background/passes.js

```diff
@@ -126,4 +128,24 @@ export class Passes extends Component {
     log("Force a new authentication for un update");
     await this.sendMessage("authenticationNeeded");
   }
+
+  async maybeScheduleUnlimitedSurvey() {
+    const {totalPasses} = await browser.storage.local.get("totalPasses");
+    if (totalPasses === undefined || !this.syncAreUnlimited()) {
```

Comment:
`this.syncAreUnlimited()` already checks for undefined so you could potentially remove that check and remove the line above.

Reply:
Right, but we have this: https://github.com/mozilla/secure-proxy/blob/master/src/background/passes.js#L34-L38 and because of this, totalPasses after init() is not undefined. Non authenticated users are treated as unlimited. This is fine for the rest of the codebase, but not here.